enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,964 @@
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdBackup", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "singlenamespace-alpha",
+                "upgradeInfo": {}
+            },
+            {
+                "name": "clusterwide-alpha",
+                "upgradeInfo": {}
+            }
+        ],
+        "relatedImages" : [ 
+            {
+                "name": "etcdv0.9.0",
+                "image": "quay.io/coreos/etcd-operator@sha256:db563baa8194fcfe39d1df744ed70024b0f1f9e9b55b5923c2f3a413c44dc6b8"
+            }
+        ]
+    },
+    {
+        "schema": "bundle.v1",
+        "name": "etcdoperator.v0.9.2",
+        "package": "etcd",
+        "image": "quay.io/operatorhubio/etcd:v0.9.2",
+        "version": "0.9.2",
+        "properties":[
+            {
+                "name": "olm.package",
+                "values": {
+                    "packageName": "etcd",
+                    "version": "0.9.2"
+                }
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdRestore", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "singlenamespace-alpha",
+                "upgradeInfo": {
+                    "replaces": "etcdoperator.v0.9.0"
+                }
+            }
+            
+        ],
+        "relatedImages":[
+            {
+                "name": "etcdv0.9.2",
+                "image": "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2"
+            }
+        ]
+    },
+    {
+        "schema": "bundle.v1",
+        "name": "etcdoperator.v0.9.2-clusterwide", 
+        "package": "etcd",
+        "image": "quay.io/operatorhubio/etcd:v0.9.2-clusterwide",
+        "version": "0.9.2-clusterwide",
+        "properties":[
+            {
+                "name": "olm.package",
+                "values": {
+                    "packageName": "etcd",
+                    "version": "0.9.2-clusterwide"
+                }
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdBackup", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "clusterwide-alpha",
+                "upgradeInfo": {
+                    "replaces": "etcdoperator.v0.9.0"
+                    //"skiprange": >=0.9.0 <0.9.2  // an example of skiprange if skiprange was specified for this bundle
```

Comment:
>Does this mean that the skiprange only applies to other bundles in the same channel?

Shouldn't that be the case ideally? Can we assume that when a bundle is built that mentions for skips/skipsrange for one channel, it would need to stick to those mandates for all future channels it's included in? With the config I feel like we have the opportunity to relax that rule, where one can opt to have the bundle skip/skiprange in every channel it's included in, or opt to behave in different ways in different channels.

Reply:
for now i expect most people to be using it to allow upgrades from one channel to another.  While i can envision that someone might want the ability to only allow upgrading within the channel and actually want to ensure that the skipRange does *not* allow someone to jump from one channel to another, that seems somewhat less likely to me (I guess maybe it'll become common when people have preview channels and they explicilty do not want you to be able to upgrade from some version in the preview channel, to a stable version, because they insist you only install stable versions if you want to be on the stable channel).

so i guess i've talked myself in circles, but i think we should start by retaining the existing behavior, and then in the future we can consider a channel-constrained skiprange if we see a need.  So let's make sure the json structure reflects that behavior (applies across channels)